feature/gribi/otg_tests/mpls_compliance/gribi_mpls_compliance_test.go

```diff
@@ -52,3 +52,74 @@ func TestMPLSLabelPushDepth(t *testing.T) {
 		})
 	}
 }
+
+// TestMPLSPushToIP validates the gRIBI actions that are used to push N labels onto
+// an IP packet. Note that this test does not validate against the dataplane.
+func TestMPLSPushToIP(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	gribic := dut.RawAPIs().GRIBI(t)
+	c := fluent.NewClient()
+	c.Connection().WithStub(gribic)
+
+	_ = mplsutil.PushBaseConfigs(t, ondatra.DUT(t, "dut"), ondatra.ATE(t, "ate"))
+
+	baseLabel := 42
+	numLabels := 20
+	for i := 1; i <= numLabels; i++ {
+		t.Run(fmt.Sprintf("push %d labels to IP", i), func(t *testing.T) {
+			mplsutil.PushToIPPacket(t, c, deviations.DefaultNetworkInstance(dut), baseLabel, i, sleepFn)
```

Comment:
i feel like this really pushs a bad boundary on hiding what the critical section of the test 
this is very unobivous what this magic function is doing

I would much rather see it structured as 

Build Config
Push Config
Start Traffic
Check Results

inside the t.Run

that makes it very clear and you can even add the right t.Log steps to show what is actually happening

Reply:
I'm not clear with why this being in a different function makes things more readable -- it's a function call, like we have other function calls in the code - yes, it implements some of the test logic, but this aids reusability of the same code. See the explanation above.